spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java

```diff
@@ -152,4 +174,45 @@ public String manifest(@PathVariable("name") String name, @PathVariable("version
 	public Collection<Deployer> platformList() {
 		return this.streamService.platformList();
 	}
+
+	/**
+	 * {@link org.springframework.hateoas.ResourceAssembler} implementation that converts
+	 * {@link StreamDeployment}s to {@link StreamDeploymentResource}s.
+	 */
+	class Assembler extends ResourceAssemblerSupport<StreamDeployment, StreamDeploymentResource> {
+
+		private final StreamDefinition streamDefinition;
+
+		public Assembler(StreamDefinition streamDefinition) {
+			super(StreamDeploymentController.class, StreamDeploymentResource.class);
+			this.streamDefinition = streamDefinition;
+		}
+
+		@Override
+		public StreamDeploymentResource toResource(StreamDeployment stream) {
+			try {
+				return createResourceWithId(this.streamDefinition.getName(), stream);
+			}
+			catch (IllegalStateException e) {
+				logger.warn("Failed to create StreamDeploymentResource. " + e.getMessage());
+			}
+			return null;
+		}
+
+		@Override
+		public StreamDeploymentResource instantiateResource(StreamDeployment stream) {
+			final StreamDeploymentResource resource = new StreamDeploymentResource(this.streamDefinition.getName(),
+					ArgumentSanitizer.sanitizeStream(this.streamDefinition.getDslText()), stream.getDeploymentProperties()
+			);
+			Map<StreamDefinition, DeploymentState> streamDeploymentStates = StreamDeploymentController.this.streamService.state(Arrays.asList(this.streamDefinition));
```

Comment:
As with the previous comment we should free the Assembler of an business logic. E.g. The the Assembler should not be aware of any business logic or domain class relationships.  It's sole purpose is to turn a single domain class (StreamDeployment in this case) into a Resource

Reply:
Yeah, this technical debt is a long due as you would notice similar pattern in some other resource assemblers as well. I will try to fix it for this assembler.